src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -298,30 +301,35 @@ private boolean isTerminated(final DetailAST ast, boolean useBreak,
                 terminated = true;
                 break;
             case TokenTypes.LITERAL_BREAK:
-                terminated = useBreak;
+                terminated = useBreak
+                    || ast.getChildCount() == 2 && !labels.contains(ast.getFirstChild().getText());
```

Comment:
Why isn't it enough to just check if a given `break` is breaking to a label? Why do we need label names?

Reply:
it is needed to handle weirdest cases like
```
        case 1:
            inner: inner2: switch (b) {
            case 10:
                break inner;
            default:
                return 11;
            }
        case 2: // violation 'Fall through from previous branch of the switch statement.'
```

In such case we need to collect label names and check if break is used with label that is outside of case block

Reply:
Leaving this unresolved so that others can see it.

Reply:
infinite loop is outside of this Check targets.